### PR TITLE
Task completed: GP-297 Improve attacking state

### PR DESCRIPTION
**Pull request by frnzee**
- changed attacking state logics
- changed character's look direction so that they look from their point of view and not from zero point
- added weapon functionality to once enemy spawner
- added check of sphere collider existing to get rid of errors on play mod stopping

### CODE REVIEW

File: Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs [hunk 1]
```diff
@@ -113,6 +119,11 @@ public void ActivateAttackCollider()
 
         public void DeactivateAttackCollider()
         {
+            if (!_sphereCollider)
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: Change _sphereCollider to "this"

File: Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs [hunk 2]
```diff
@@ -105,6 +106,11 @@ private void ProcessRotation(bool isMoving)
 
         public void ActivateAttackCollider()
         {
+            if (!_sphereCollider)
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: Change _sphereCollider to "this"

File: Assets/Scripts/Gameplay/Enemy/Spawners/OnceEnemySpawner.cs [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
         {
-            Spawn();
+            SpawnOne();
             Destroy(gameObject);
         }
+        
+        private void SpawnOne()
+        {
+            var enemy = Spawn();
+            enemy.ChangeWeapon(_weaponFactory.Create(_weaponSettings, enemy));
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Enemy/Spawners/OnceEnemySpawner.cs]: Move this behaviour to parent class.

**Review by ega-ega — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs (hunk 2 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: done

File: Assets/Scripts/Gameplay/Enemy/Spawners/OnceEnemySpawner.cs (hunk 3 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/Enemy/Spawners/OnceEnemySpawner.cs]: done

**Pull request merged**